proto/frequenz/api/dispatch/v1/dispatch.proto

```diff
@@ -258,31 +258,35 @@ message DispatchFilter {
 
 // Parameter for controlling which components a dispatch applies to
 // either a set of component IDs, or a set of component categories.
+// Examples:
+// - To dispatch to a set of component IDs:
+//   selector { component_ids { ids: [1, 2, 3] } }
+// - To dispatch to a set of component categories:
+//   selector { component_categories { categories: [COMPONENT_CATEGORY_BATTERY, COMPONENT_CRYPTO_MINER] } }
 message ComponentSelector {
-  oneof selector {
+  // Wrapper for controlling dispatches with a set of component IDs
+  // Required as we can't use `repeated` directly in a `oneof`
+  message IdSet {
     // Set of component IDs
-    ComponentIDs component_ids = 1;
+    repeated uint64 ids = 1;
+  }
 
-    // Component categories
-    ComponentCategories component_category = 2;
+  // Wrapper for controlling dispatches with a set of component categories
+  // Required as we can't use `repeated` directly in a `oneof`
+  message CategorySet {
+    // Set of component categories
+    repeated frequenz.api.common.v1.microgrid.components.ComponentCategory categories = 1;
   }
-}
 
-// Wrapper for controlling dispatches with a set of component IDs
-// Required as we can't use `repeated` directly in a `oneof`
-message ComponentIDs {
-  // Set of component IDs
-  repeated uint64 component_ids = 1;
-}
+  oneof selector {
```

Comment:
I am wondering if this might be nicer:

```
oneof component_selector {
  IdSet ids = 1;
  CategorySet categories = 2;
}
```
```

Reply:
It's less verbose, but I think the extra verbosity provides some clarity here. A component ID is a more specific "thing" than just any ID. It comes with a set of assumptions, that it corresponds to a physical component, and that the IDs are unique across microgrids. Similar with component categories. So I think it's fine as it is.

Reply:
I suppose strictly speaking we should name this 
```
oneof selector {
  ComponentSet components = 1;
  CategorySet categories = 2;
}
```